public/app/features/datasources/components/EditDataSource.tsx

```diff
@@ -149,6 +151,23 @@ export function EditDataSourceView({
     );
   }
 
+  // Used by extensions that can only modify certain parts of the `jsonData`
+  const onUpdateJsonData = (jsonData: Record<string, unknown>) => {
+    const protectedFields = ['authType', 'defaultRegion', 'profile', 'manageAlerts', 'alertmanagerUid'];
+    onOptionsChange({ ...dataSource, jsonData: { ...dataSource.jsonData, ...omit(jsonData, protectedFields) } });
+  };
+
+  const { extensions } = getPluginElementExtensions({
+    extensionPointId: PluginExtensionPoints.DataSourceConfig,
+    context: { jsonData: dataSource.jsonData, onUpdateJsonData, pluginId: dataSource.type },
```

Comment:
Any chance to add onSubmit here too? Either passing nothing or dataSource.jsonData? 
This will be needed to persist data to a PDC endpoint when user hits save...

Reply:
Yepp, that's a good point! (I was thinking about doing the persisting after each onChange / onBlur, but I think it makes sense to pass an onSubmit as well.)